### ISSUE TITLE
versions in firefox, opera stores are (extremely) old

### ISSUE THREAD

**Issue by errorgurumeditation** [ray-lothian/UserAgent-Switcher#222]
the version in firefox ~~and edge~~ store~~s~~ is 0.5.0, from april 17, 2023, and the version in the opera store is 0.3.2, from august 1, 2019. only the version in chrome webstore is up-to-date

edit: seems version in edge store has been updated

**Comment by YesTrustMe** [comment]
I wonder if the extension has been abandoned. Perhaps someone can create a fork of it and post in the app store. I can never get the FF extension files to install manually simply.

**Comment by YesTrustMe** [comment]
I wonder if the extension has been abandoned on FF. More Chrome users will be switching to FF in 2025 due to Google removing MV2 extensions.

Perhaps someone can create a fork of it and post in the app store. I can never get the FF extension files to install manually simply or easily from Github eg. Bypass Firewall and others.

**Comment by Hello-World-Traveler** [comment]
> I wonder if the extension has been abandoned on FF. More Chrome users will be switching to FF in 2025 due to Google removing MV2 extensions.
> 
> Perhaps someone can create a fork of it and post in the app store. I can never get the FF extension files to install manually simply or easily from Github eg. Bypass Firewall and others.

@YesTrustMe try this to install addons. This might work for this Addon if you choose V2 or V3.

1.     Compress the extracted folder into a ZIP file
2.     Rename the ZIP file extension from ".zip" to ".xpi"
3.     In Firefox, go to "about:addons"
4.     Click the gear icon and select "Install Add-on From File"
5.     Choose the .xpi file you created

@ray-lothian Not to rush you in anyway, Will this Addon be update in the FireFox Store soon?

**Comment by Alexgopen** [comment]
Any update?

**Comment by timur-g** [comment]
Thanks for the instruction. I could not upgrade with installing v2 in Firefox ESR 115. 
I could use `about:debugging`, click This Firefox, click Load Temporary Add-on button, load XPI. 
I could see it working by updated UA. Got Warning details:
`Reading manifest: Warning processing storage: An unexpected property was found in the WebExtension manifest.`
The extension remains installed only until we restart Firefox.

It would be good to have working instructions.

**Comment by warframe-tools** [comment]
Is help needed to update this @ray-lothian ? I noticed that the Chrome and Edge versions are getting updates regularly, but the FIrefox one is over 2 years old on updates.

**Comment by ray-lothian** [comment]
"v3" is pretty much ready for Firefox release. Give it a test and report back!

**Comment by warframe-tools** [comment]
> "v3" is pretty much ready for Firefox release. Give it a test and report back!

Seems to work, gave a warning on the debugging install (not sure if that is because of that, just started messing around with extensions)

`Reading manifest: Warning processing storage: An unexpected property was found in the WebExtension manifest.`

**Comment by ray-lothian** [comment]
Ignore the warning—it's caused by the manifest having both worker and background page entries.

**Comment by warframe-tools** [comment]
Okay. Tested with macOS and Windows on Chrome, Edge, Safari and Opera and all worked just fine to change the user agent. Google was not happy and popped a captcha, but im not sure that is something that can be avoided?

**Comment by ray-lothian** [comment]
> Google was not happy and popped a captcha, but im not sure that is something that can be avoided?

Nope, at least, this isn't related to the M3 migration. This version also adds an "Ignore Tab" context menu option to bypass websites (like Cloudflare) that don’t function properly when the extension is enabled.

**Comment by HelpfulSoft1207** [comment]
(same person, was on the wrong account)

Understood. It looks like it is ready to go then, from the limited testing I have done.

**Comment by ray-lothian** [comment]
I am updating FF to the latest version.

**Comment by HelpfulSoft1207** [comment]
How long does that normally take on AMO?
